force-app/main/default/classes/SOQL.cls

```diff
@@ -962,6 +969,10 @@ public virtual inherited sharing class SOQL implements Queryable {
             groupedFields.add(field.getDescribe().getName());
         }
 
+        public void withGroupedField(String relationshipName, SObjectField field) {
+            groupedFields.add(relationshipName + '.' + field.getDescribe().getName());
```

Comment:
We can do something like that, Apex invokes `.getDescribe().getName()` automatically.
It applies only when the String is concatenated with `SObjectField`.

`groupedFields.add(relationshipName + '.' + field);`